libs/linear/Data/Linear/Notation.idr

```diff
@@ -21,6 +21,10 @@ public export
 export prefix 5 !*
 ||| Prefix notation for the linear unrestricted modality
 public export
-record (!*) (a : Type) where
-  constructor MkBang
-  unrestricted : a
+data (!*) : Type -> Type where
+  MkBang : a -> !* a
+
+||| Unpack an unrestricted value in a linear context
+public export
+unrestricted : !* a -@ a
+unrestricted (MkBang unr) = unr
```

Comment:
I've noticed we lose `(.unrestricted)`. Not sure what's the most favourable way to handle that.

Reply:
yes, for this to be completely equivalent we need both `unrestricted` and `(.unrestricted)` might as well add both

Reply:
That would be backwards compatible, but it doesn't feel in the spirit of the stdlib

Reply:
Would this be a precedent? (I searched for `\(\.\w`)

https://github.com/idris-lang/Idris2/blob/1678b4676f619bc28bf6eea25b789d7cf7f2f9d6/libs/base/Data/Singleton.idr#L13-L19